api/scm.go

```diff
@@ -191,7 +197,43 @@ func SyncRepo(c *gin.Context) {
 
 			return
 		}
+
+		err = updateUserFavorites(c, r)
+		if err != nil {
+			util.HandleError(c, http.StatusInternalServerError, err)
+			return
+		}
 	}
 
 	c.JSON(http.StatusOK, fmt.Sprintf("repo %s synced", r.GetFullName()))
 }
+
+// updateUserFavorites is a helper function that takes a repo
+// that has been deleted from the SCM and ensures that it does
+// not appear in the user's favorites after its deletion.
+func updateUserFavorites(c *gin.Context, r *library.Repo) error {
+	// get user from the database
+	u, err := database.FromContext(c).GetUser(r.GetUserID())
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to retrieve user in database", baseErr)
+		return retErr
+	}
+
+	// if deleted repo exists in favorites, remove it
+	favorites := []string{}
+	for _, fav := range u.GetFavorites() {
+		if fav != r.GetFullName() {
+			favorites = append(favorites, fav)
+		}
+	}
+	u.SetFavorites(favorites)
+
+	// update the user favorites slice
+	err = database.FromContext(c).UpdateUser(u)
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to update repo %s: %v", baseErr, r.GetFullName(), err)
+		return retErr
```

Comment:
Thoughts on condensing this into one line?

```suggestion
		return fmt.Errorf("%s: failed to update repo %s: %v", baseErr, r.GetFullName(), err)
```